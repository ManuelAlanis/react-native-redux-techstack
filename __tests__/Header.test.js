
import 'react-native';
import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import Header from '../src/components/header';
import {
    View,
    TouchableOpacity, 
    Text,
    StatusBar
} from 'react-native';

const PROGRESS_TEST = 90;

describe('Unit test for header views and functions', () => {
    it('test renders correctly', () => {
        const tree = renderer.create(<Header />);
        expect(tree).toMatchSnapshot();
    });

    it('comprobe increment of progress value over state context', () =>{
        const wrapper = renderer.create(
            <Header />
        );

        const instance = wrapper.getInstance();
        const handleIncreaseProgress = jest.spyOn(instance, 'increaseProgress');
        instance.increaseProgress(PROGRESS_TEST);

        expect(instance.state.progress).toEqual(PROGRESS_TEST);
        expect(handleIncreaseProgress).toHaveBeenCalled();
    });

    it('test addTwoNumbers add numbers correctlty', () => {
        const response = new Header().addTwoNumbers(5,10);
        expect(response).toEqual(5+10);
    });

    it('test addTwoNumbers not return null', () => {
        const response = new Header().addTwoNumbers(5,10);
        expect(response).not.toEqual(null);
    });

    it('test addTwoNumbers not return toBeFalsy', () => {
        const response = new Header().addTwoNumbers(5,10);
        expect(response).not.toBeFalsy();
    });

    // import mockJSON_RESPONSE from '..libraryMock'

    // it('test addTwoNumbers not return Truthy', () => {
    //     const response = await new getTimeService().addTwoNumbers(5,10);

    //     if(response.success) {
    //         expect(response.firstName).not.toEqual('');
    //         expect(response.firstName).not.toEqual('');
    //         expect(response.firstName).not.toEqual('');
    //         expect(response.firstName).not.toEqual('');
    //     }
    // });

    it('test found 1 StatusBar element', () =>{
        const wrapper = renderer.create(<Header />);
        const instance = wrapper.root;
        expect(instance.findAllByType(StatusBar)).toHaveLength(1);
    });
});