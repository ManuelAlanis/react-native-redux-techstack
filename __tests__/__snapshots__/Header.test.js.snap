// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for header views and functions test renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "backgroundColor": "#2626ed",
        "color": "#fff",
        "fontSize": 20,
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingTop": 10,
        "textAlign": "center",
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Press here to kill you...
    </Text>
  </View>
</View>
`;
